plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/button/frontend.tsx

```diff
@@ -144,6 +146,14 @@ const { state } = store< Store >( 'woocommerce/product-button', {
 					quantityToAdd
 				);
 
+				// Add a manual trigger of the event for backward compatibility
+				// and symmetry (remove event is triggered properly from Cart and Mini Cart).
+				// We'll try to unify the events in future.
+				yield doAction(
+					`experimental__woocommerce_blocks-cart-add-item`,
+					product
```

Comment:
Is there a particular reason why you didn't use `dispatchStoreEvent`? ([example](https://github.com/woocommerce/woocommerce/blob/ddd877441c74da8f128811e08405f8ca8c681c42/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/button/block.tsx/#L75-L77)).

Reply:
@gigitux

Reply:
@gigitux 
Yes, so the problem is that in order to use `dispatchStoreEvent` you need to use`useStoreEvents` which is a [hook based on `useCallback` hook](https://github.dev/woocommerce/woocommerce/blob/ddd877441c74da8f128811e08405f8ca8c681c42/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/button/block.tsx/). It cannot be used outside of React component context and in this case we don't satisfy that need.

I was thinking of creating a non-hook helper function that would be simply the same as `dispatchStoreEvent` wrapping `doAction` but considering we're thinking of unifying and potentially cleaning the events up I reckoned I can add a single instance.

What do you think? Should I prepare a helped function for this? Also, I'm going to create a followup issue to cleanup and unify the events.

Reply:
@kmanijak Good point! I suggest opening a new issue and adding a comment in the code that contains the link to the issue.

Thanks for taking care of this! LGTM!

Reply:
Issue created: https://github.com/woocommerce/woocommerce/issues/42985 and referenced in the comment 👍